### DISCUSSION TITLE
Syncing data across devices

### DISCUSSION THREAD

**Discussion by jonjomckay** [jonjomckay/fritter#262] [Ideas]
After hearing some common feedback from a few people, I'm planning to add support for syncing your Fritter subscriptions, groups, saved tweets and settings across devices.
          
Naturally, this functionality would be included in the app, but would need to depend on a backend/API running somewhere. I would 100% be releasing this backend service as an open-source web application that anyone can self-host, as I know privacy is people's number 1 priority when it comes to using Fritter 🕵🏽 

I do also plan on trying to support the Android [Storage Access Framework](https://developer.android.com/guide/topics/providers/document-provider), but this does have some limitations (including Flutter-imposed ones). It also wouldn't work on iOS devices when I finally get around to enabling those builds.

I'd also like to gather interest in syncing using a cloud instance of this service, for a nominal, user-chosen price (~$0.50 a month minimum, to cover fees). I realise a lot of people aren't in a position to self-host something, so this would be a convenience to them, and would double as a way of supporting the project.

Please let me know your thoughts, ideas and concerns in this thread, and we'll get discussing 🙂

**Comment by cobratbq** [comment]
I currently use syncthing and export my data periodically. I am aware that this is not the same as keeping two devices/apps in sync at all times. I just want to mention that this is an option right now, given that synchronization comes with its own set of difficulties.

FYI, another approach to "eventually synchronized" would be to have a secondary store in a file and periodically comparing the file content to current fritter state. The file could be in a Dropbox/Box/Syncthing synchronized folder. It is not as fancy as real-time synchronization, but it is an option, and especially with Syncthing the user has full control.
